koku/api/report/queries.py

```diff
@@ -669,11 +669,13 @@ def _get_tag_group_by(self):
         """Create list of tag based group by parameters."""
         group_by = []
         tag_groups = self.get_tag_group_by_keys()
+        sanitize_pattern = re.compile(r'[\s"\'`;]+')
```

Comment:
For situations like this, a translation table is simpler and more performant, plus it doesn't bring the maintenance overhead of regular expressions.
```suggestion
        characters_to_sanitize = r' "\'`;'
        table = str.maketrans(characters_to_sanitize, "_" * len(characters_to_sanitize))
        ...
        sanitized_tag = strip_prefix(tag, TAG_PREFIX).translate(table)
```